### PR TITLE
Refactor resolving of null information for custom casted attribute types

### PR DESCRIPTION
**Pull request by wimski**
## Summary

Let's say we have the following custom cast:

```php
class MyCustomCast implements CastsAttributes
{
    /**
     * @param Model                $model
     * @param string               $key
     * @param mixed                $value
     * @param array<string, mixed> $attributes
     * @return SomeObject|null
     */
    public function get($model, string $key, $value, $attributes): ?SomeObject
    {
        if ($value === null) {
            return null;
        }

        return new SomeObject($value);
    }

    // setter omitted from example for brevity
}
```

 Now suppose we have two pretty much identical models which use this cast:

```php
class ModelA extends Model
{
    protected $table = 'models_a';

    /**
     * @var array<string, string>
     */
    protected $casts = [
        'casted_column' => MyCustomCast::class,
    ];
}
```

```php
class ModelB extends Model
{
    protected $table = 'models_b';

    /**
     * @var array<string, string>
     */
    protected $casts = [
        'casted_column' => MyCustomCast::class,
    ];
}
```

These models have tables created with the following migrations:

```php
Schema::create('models_a', function (Blueprint $table) {
	$table->id();
	$table->string('casted_column');
	$table->timestamps();
});
```

```php
Schema::create('models_b', function (Blueprint $table) {
	$table->id();
	$table->string('casted_column')->nullable();
	$table->timestamps();
});
```

The only difference between A and B is the `nullable()` setting on the column. Based on that difference I would expect the following properties to be written by the IDE helper:

```php
/**
 * @property SomeObject $casted_column
 */
class ModelA extends Model
```

```php
/**
 * @property SomeObject|null $casted_column
 */
class ModelB extends Model
```

However, both models will receive `SomeObject|null` as the property type, because currently the IDE helper only check the cast class without taking the migration into account. In my opinion the migration should inform the possibility of a `null` value and not the cast, because the cast doesn't know anything about the column it's going to be used for and should always implement `X|null`.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

In my opnion this is a fix for behaviour that was always incorrect, but because it changes behaviour (and outcome), I do consider it a breaking change. I'll let it up to the maintainers how they want to deal with this.

### Checklist
- [x] Existing tests have been adapted and/or new tests have been added
- [ ] Add a CHANGELOG.md entry
- [ ] Update the README.md
- [x] Code style has been fixed via `composer fix-style`

### CODE REVIEW

File: tests/Console/ModelsCommand/LaravelCustomCasts/__snapshots__/Test__test__1.php [hunk 1]
```diff
@@ -28,8 +28,9 @@
  * @property \Barryvdh\LaravelIdeHelper\Tests\Console\ModelsCommand\LaravelCustomCasts\Casts\CastedProperty $casted_property_with_return_docblock
  * @property \Barryvdh\LaravelIdeHelper\Tests\Console\ModelsCommand\LaravelCustomCasts\Casts\CastedProperty $casted_property_with_return_docblock_fqn
  * @property array $casted_property_with_return_primitive
- * @property array|null $casted_property_with_return_primitive_docblock
- * @property array|null $casted_property_with_return_nullable_primitive
+ * @property array $casted_property_with_return_primitive_docblock
+ * @property array $casted_property_with_return_nullable_primitive
```
Comment (mfn) [inline tests/Console/ModelsCommand/LaravelCustomCasts/__snapshots__/Test__test__1.php:32]: Why is this changed? Is it intentional based on the fix? Thanks

Comment (wimski) [reply tests/Console/ModelsCommand/LaravelCustomCasts/__snapshots__/Test__test__1.php:32]: Yes, this is intentional. I didn't change the migration for these properties and they weren't marked as `nullable()`, so the `|null` gets removed by the fix in this PR.

**Comment by daniel-de-wit** [comment]
@mfn Can you please take another look at this?

**Comment by barryvdh** [comment]
Added to changelog in https://github.com/barryvdh/laravel-ide-helper/commit/aa1aa66ee6cd70acb6c8be6cc0fc2bdee3dc76ba
Thanks!
